src/main/java/me/jellysquid/mods/lithium/mixin/world/chunk_inline_block_access/WorldChunkMixin.java

```diff
@@ -47,10 +49,19 @@ public BlockState getBlockState(BlockPos pos) {
 
     /**
      * @reason Reduce method size to help the JVM inline
-     * @author JellySquid
+     * @author Maity
      */
     @Overwrite
-    public FluidState getFluidState(BlockPos pos) {
-        return this.getBlockState(pos).getFluidState();
+    public FluidState getFluidState(int x, int y, int z) {
+        // WorldChunk#getFluidState(BlockPos) redirects to this method
```

Comment:
What is the improvement here? The order in which methods are called is a bit different, and marginally speeding up the case of getting blocks outside the world does not seem necessary unless it is actually an often taken branch.

Reply:
> What is the improvement here? The order in which methods are called is a bit different, and marginally speeding up the case of getting blocks outside the world does not seem necessary unless it is actually an often taken branch.

`getFluidState(BlockPos)` redirects to `getFluidState(int, int, int)` when Lithium redirects it to the optimized version, but `getFluidState(int, int, int)` remained non-optimized. This change will optimize all methods.

P. S. I hope you understand, because I am from another country and I can make stupid mistakes in English.

Reply:
The only place where `getFluidState(int, int, int)` is called is `getFluidState(BlockPos)` (and maybe some mods also call it)

Reply:
> The only place where `getFluidState(int, int, int)` is called is `getFluidState(BlockPos)` (and maybe some mods also call it)

Well, that's not bad, is it?